src/main/java/net/minecraftforge/common/util/EnumHelper.java

```diff
@@ -250,6 +215,8 @@ private static void cleanEnumCache(Class<?> enumClass) throws Exception
     //Tests an enum is compatible with these args, throws an error if not.
     public static void testEnum(Class<? extends Enum<?>> enumType, Class<?>[] paramTypes)
     {
+        if (!(EnumHelper.class.getClassLoader() instanceof LaunchClassLoader))
```

Comment:
You should throw an exception since once this is loaded, all future operations will break for sure.

Reply:
This is to allow the tests to run. If Minecraft itself is running under it, nothing is going to work

Reply:
Then this thing is generally skipping tests because you just modified one test to run under it

Reply:
No, it will still fail when an addEnum is called if it is not running in the correct classloader. This only affects static initialisers

Reply:
> This only affects static initialisers

What doe this mean?

Reply:
it is only used in `static{}` blocks. These are loaded in various tests that don't actually use `EnumHelper`